### ISSUE TITLE
Problems with NamedUser and membership

### ISSUE THREAD

**Issue by lbrownell-gpsw** [PyGithub/PyGithub#507]
Here's what I'm doing.  All's well until I try either method to add a user.

Definitions:

- args.org:  The organization name; the team and the user are both in this org.
- args.team: The team name to add the user as a member
- args.token: Github oauth token.
- args.user:  The login name of the user to add.  This user is a member of the org, but not yet of the team -- which is what I'm trying to do, add the user to the team.  Note that for memberships, it shouldn't matter if the user is a member of the org or not, as it will invite the user to the org and, if accepted, add them to the team.  Very handy that.
```
from github import Github
gh = Github(token)
org = gh.get_organization(args.org)
team = None
# Find the team ID.
for t in org.get_teams():
    if t.name == args.team:
        team = t
        break

if team:
    team.add_membership(args.user)
    team.add_to_members(args.user)
```
In each case, I get the following:
```
Traceback (most recent call last):
  File "addUsersToTeamInOrg.py", line 52, in <module>
    team.add_to_members(args.user)
  File "C:\Python27\lib\site-packages\github\Team.py", line 122, in add_to_members
    assert isinstance(member, github.NamedUser.NamedUser), member
AssertionError: joe-user
```
Note that "joe-user" isn't the user login used, but the one that was used is valid and should've worked. Directly calling the APIs did work, in fact.

In the github API documentation, you supply the login ID for `username`:
```
PUT /teams/:id/members/:username
PUT /teams/:id/memberships/:username
```

One other thing:  For `memberships`, there is an optional `role` parameter, which you also don't seem to support.

Thanks.

**Comment by lbrownell-gpsw** [comment]
Oh hell, nevermind...mostly.  You pass the user object, not the login name when `username` is used.  Duh.

However, that last thing...is the `role` parameter for memberships working?

**Comment by lbrownell-gpsw** [comment]
But...how to you get a` NamedUser`?  If I have a `username` (e.g., their login ID), how can I get a NamedUser object for that, without search thru an organization or team?  E.g., what's your implementation for `GET /users/:username`?  If you want to use the `membership` action to invite someone from outside the organization, searching for them won't work.

**Comment by lbrownell-gpsw** [comment]
Damn...so, yeah just get_user(username) does the trick.

I'm lame, and your docs are lacking...it's a tough combo.  :-)

However...still, please look into the `membership` `role` parameter.

**Comment by bbriggs** [comment]
+1 for showing me how to get a NamedUser ;)

**Comment by suranc** [comment]
@lbrownell-gpsw I had an use case for the `membership` `role` parameter, and added it in pull request #602 

However, I have just opened the pull request, so it has not yet been merged.

**Comment by sfdye** [comment]
Fixed in #638
